### PR TITLE
Feature/get spending query

### PR DESCRIPTION
**Pull request by anastasiya-nd**

### CODE REVIEW

**Comment by anastasiya-nd** [comment]
@ok2ju 
I have changed code for this pull request.

File: src/components/NewSpendingForm/NewSpendingForm.jsx [hunk 1]
```diff
@@ -69,10 +87,23 @@ const NewSpendingForm = ({ onClose }) => {
...
+      currency,
+      note,
+      labels,
+    };
     if (!fields.category || !fields.createdAt || !fields.amount || !fields.currency) {
       console.log('Required fields');
     } else {
-      dispatch(postSpendingPending(fields));
+      if (spending) {
+        fields._id = spending._id; //eslint-disable-line
```
Comment (ok2ju) [inline src/components/NewSpendingForm/NewSpendingForm.jsx:102]: Why you set id to the fields? You use `PATCH` method on back end site. It means that to update entity we don't need to provide a full list of object properties (only that changed).

File: src/redux/reducer.js [hunk 2]
```diff
@@ -71,6 +74,28 @@ const reducer = (state = initialState, action) => {
...
+    case UPDATE_SPENDING_SUCCESS:
+      return {
+        ...state,
+        spendings: state.spendings.map((spending) => {
+          if (spending._id === action.payload.id) { //eslint-disable-line
+            return { ...action.payload.spending };
+          }
+          return spending;
+        }),
+        spending: action.payload.spending,
```
Comment (ok2ju) [inline src/redux/reducer.js:91]: Do you use this field?

File: src/redux/sagas.js [hunk 3]
```diff
@@ -47,10 +54,20 @@ function* postSpendingWorker(data) {
   }
 }
 
+function* updateSpendingWorker(action) {
+  try {
+    yield call(spendings.updateSpending, action.payload.id, action.payload.spending);
+    yield put(updateSpendingSuccess(action.payload.id, action.payload.spending));
```
Comment (ok2ju) [inline src/redux/sagas.js:60]: Backend should return to the frontEnd updated spending and that spending you should store.

File: src/components/NewSpendingForm/NewSpendingForm.jsx [hunk 4]
```diff
@@ -22,21 +23,38 @@ const NewSpendingForm = ({ onClose }) => {
...
     { id: '4', value: 'eur', label: 'EUR' },
   ];
-  const [category, setCategory] = useState(''); //eslint-disable-line
+  const [category, setCategory] = useState('');
   const [createdAt, setCreatedAt] = useState('');
   const [amount, setAmount] = useState(0);
   const [currency, setCurrency] = useState('');
   const [note, setNote] = useState('');
   const [labels, setLabels] = useState([]);
+  const spending = useSelector(getSpendings).find((item) => item._id === id); //eslint-disable-line
```
[the remark this answers is not in the document]
Comment (ok2ju) [reply src/components/NewSpendingForm/NewSpendingForm.jsx:32]: Also, when component mount create request for the spending by id from props (and place this spending along with other spendings in array).

**Review by ok2ju — CHANGES_REQUESTED** [review]

File: src/redux/sagas.js (hunk 3 above)
Comment (anastasiya-nd) [reply src/redux/sagas.js:60]: > Backend should return to the frontEnd updated spending and that spending you should store.

Is it right?

```
const updateSpending = async (req, res) => {
  try {
    const { ...values } = req.body
    const id = req.params.id
    const spending = await Spending.findByIdAndUpdate(id, `values)`
    res.status(HTTP_STATUS_CODES.OK).send(spending)
  } catch (error) {
    res
      .status(HTTP_STATUS_CODES.INTERNAL_SERVER_ERROR)
      .send({ error: 'Internal Error - Internal server error' })
  }
}
```

Comment (anastasiya-nd) [reply src/redux/sagas.js:60]:
```
function* updateSpendingWorker(action) {
  try {
    const response = yield call( spendings.updateSpending, action.payload.id,  action.payload.spending );
    yield put(updateSpendingSuccess(action.payload.id, response.data));
  } catch (error) {
    yield put(updateSpendingError());
  }
}
```

**Review by ok2ju — CHANGES_REQUESTED** [review]

**Review by ok2ju — APPROVED** [review]

**Pull request merged**